src/openforms/contrib/zgw/service.py

```diff
@@ -46,37 +48,32 @@ def create_zaak(
 
     data.update(extra)
 
-    zaak = client.create("zaak", data)
+    zaak = zrc_client.create("zaak", data)
     return zaak
 
 
-def default_get_drc() -> Service:
-    config = ZgwConfig.get_solo()
-    return config.drc_service
-
-
-def partial_update_zaak(zaak_url: str, data: dict) -> dict:
-    config = ZgwConfig.get_solo()
-    client = config.zrc_service.build_client()
-    zaak = client.partial_update("zaak", data, url=zaak_url)
+def partial_update_zaak(zrc_client: ZGWClient, zaak_url: str, data: dict) -> dict:
+    zaak = zrc_client.partial_update("zaak", data, url=zaak_url)
     return zaak
 
 
-def set_zaak_payment(zaak_url: str, partial: bool = False) -> dict:
+def set_zaak_payment(
+    zrc_client: ZGWClient, zaak_url: str, partial: bool = False
+) -> dict:
     data = {
         "betalingsindicatie": "gedeeltelijk" if partial else "geheel",
         "laatsteBetaaldatum": timezone.now().isoformat(),
     }
-    return partial_update_zaak(zaak_url, data)
+    return partial_update_zaak(zrc_client, zaak_url, data)
 
 
 def create_document(
     name: str,
     base64_body: str,
     options: dict,
-    get_drc=default_get_drc,
+    get_drc: Callable,
```

Comment:
FYI, you can also specify the types of args and return type:

```suggestion
    get_drc: Callable[[], ZGWClient],
```
-> no args, `ZGWClient` return type